app/controllers/home_controller.rb

```diff
@@ -1,9 +1,12 @@
 class HomeController < ApplicationController
   def index
-    @projects = if params[:all] == "true" 
-      Project.all
-    else
-      current_user.projects unless current_user.nil?
-    end
+    @projects = if params[:all] == "true"
+                  policy_scope(Project.all)
+                elsif signed_in?
+                  policy_scope(current_user.projects)
+                else
+                  skip_policy_scope
+                  []
```

Comment:
Itt gyanús nekem, hogy szükségünk van a `signed_in?` helperre.
A feladat így szólt: "[...] the project, but every logged in user should be able to see them."

Tehát be nem jelentkezett felhasználónak nem szabadna projekteket listáznia. Ez a logika nem a jelenlegi kontroller akcióhoz kapcsolódik, hogy egy a projektre általánosságában megfogalmazható szabály. Ilyen esetben érdemes ezt a logikát átmozgatni a `project_policy.rb`-be.

pl.:

```
class AsdPolicy < ApplicationPolicy
 # ...
  class Scope < Scope
    def resolve
      if user
        scope.where(user_id: user.id)
      elsif another_helper_probably_defined_in_application_policy_returning_a_boolean?
        scope.all
      else
        scope.none
      end
    end
  end
end
```

Reply:
Ha az jó akkor az ApplicationPolicy initialize-jába, és az ottani Scope-ba is bele lehet szőni a bejelentkezés check-et. Azért nem láttam eddig értelmét ezzel foglalkozni, mert ha nincs bejelentkezve senki akkor meg se jeleni az opció, projekteket lásson vagy bármit kezdjen velük.

Reply:
Az ApplicationPolicy-t ne írjuk át. Az így szuper ahogy most van.
Az a gondolatmenet, hogy ha default-olni kell az ApplicationPolicy-re, akkor nem fér hozzá (és nem lát semmit) semmihez a felhasználó.
A Policy-k nem baj, ha repetitívek. Az a dolguk, hogy szigorúak legyenek, és ne engedjünk meg véletlenül"" valamit a felhasználónak, csak azért, mert kódot akartunk spróolni.

Reply:
Láttam a többi kommentet. Ha máshol sem vizsgálunk rá külön, hogy be van-e jelentkezve a felhasználó, akkor itt sem kell külön. Elég ha megszűnteted az `else` ágat.